specutils/spectra/spectrum_mixin.py

```diff
@@ -208,14 +215,27 @@ def with_spectral_unit(self, unit, velocity_convention=None,
                      even if your spectrum has air wavelength units
 
         """
-        new_wcs, new_meta = self._new_spectral_wcs(
-            unit=unit,
-            velocity_convention=velocity_convention or self.velocity_convention,
-            rest_value=rest_value or self.rest_value)
 
-        spectrum = self.__class__(flux=self.flux, wcs=new_wcs, meta=new_meta)
+        velocity_convention = velocity_convention if velocity_convention is not None else self.velocity_convention # noqa
+        rest_value = rest_value if rest_value is not None else self.rest_value
+        unit = self._new_wcs_argument_validation(unit, velocity_convention,
+                                                 rest_value)
+
+        # Store the original unit information and WCS for posterity
+        meta = self._meta.copy()
```

Comment:
```suggestion
        meta = self._meta.deepcopy()
```
This would also be needed if `meta` itself includes nested objects (see test example)